### DISCUSSION TITLE
Fetch Balance on different chain than the one connected

### DISCUSSION THREAD

**Discussion by velea18** [NoahZinsmeister/web3-react#466] [General]
Hello. I have a question, is it possible to fetch the native ETH balance of any other chain than the one you are connected? Like, I want to fetch my BSC balance while connected to ETH mainnet. Is it possible?

**Comment by agcty** [comment]
Yes this can be done in a number of ways

* use 2 providers
* use an API

I would go with the provider approach as you don't have to pay for an additional service.

In this case, I'd probably connect to the mainnet account with metamask or walletconnect or some other connector and simply use a StaticJsonRPCProvider for fetching the BSC balance with contract.balanceOf. However, note that the BSC account won't have a signer attached so you can't really sign and execute transactions, you can however read from the chain.

**Comment by velea18** [comment]
No, i just want to read actually. I want to build a zapper.fi like app, where you can track your balance on multiple chains

**Comment by velea18** [comment]
I see that I can accomplish this with ethers.js. But using a third party rpc like infura. is it possible to use the web3-react core instead of the bare ethers.js library?

**Comment by velea18** [comment]
i use something like this to get the priority connector which the user choose: metamask, wallet connect, etc.:

`export const useActiveConnector = (): Web3ReactPriorityHooks => {
    return getPriorityConnector(...Object.values(SUPPORTED_WALLETS).map((walletInfo): [Connector, Web3ReactHooks] => {
        return [walletInfo.connector, walletInfo.hooks]
    }));
}`

then when I need to access some info :

` const { usePriorityAccount, usePriorityWeb3React, usePriorityProvider } = useActiveConnector();

    const account = usePriorityAccount();
    const { library } = usePriorityWeb3React(usePriorityProvider());`

I assume if I want to use the web3-react hooks I just need to instantiate a number of web3-providers for each chain I m gonna use then, when using usePriorityWeb3React, just pass another provider. The issue is that, that hook accepts only a web3 provider, not a jsonrpc one

`const chainData = getChainParameters(chainId ?? 1);

    const bscProvider = new ethers.providers.JsonRpcProvider(RPC[ChainId.BSC])
    const ethProvider = new ethers.providers.JsonRpcProvider(RPC[ChainId.ETHEREUM]);

    if (window.ethereum && account) {
        const web3P = new ethers.providers.Web3Provider(window.ethereum, ethers.providers.getNetwork(ChainId[ChainId.BSC_TESTNET]))
    }
`

here I m able to read chain data from bsc and eth, but if I want to instantiate a web3provider using ethers, I cant on a chain different than the one I m connected using metamask.

If someone can give me a direction with this issue, I would appreciate very much :)

**Reply by javierguzman** [reply]
Hello @velea18 did you achieve something at the end? I want to achieve the same as well and I use ethers. As far as I understand Web3provider can only connect the current selected chain though I can use the RPC provider. The problem I have with the RPC provider is that I would need to specify the URL with the API key for example from Alchemy and that information would be in the client side which means is accessible to everybody. If you have any insight please let me know. Thank you very much and regards
